### ISSUE TITLE
AlignAfterOpenBracket: BlockIndent splits cast operator in a non-standard way

### ISSUE THREAD

**Issue by bradlitterell** [llvm/llvm-project#57250]
With clang-format version 14.0.5

`AlignAfterOpenBracket:BlockIndent` splits a cast operator in a non-standard way:

Example: 
The default "Align" style leaves this unchanged:
```console
$ clang-format -i -style="{BasedOnStyle: LLVM, AlignAfterOpenBracket: Align}" testfmt.c
int foo(void) {
  int x = (int)(100000 + 200000 + 100000 + 200000 + 100000 + 200000 + 100000 +
                200000 + 100000 + 200000 + 100000 + 200000 + 100000 + 200000 +
                100000 + 200000 + 100000 + 200000);
  int y = 100000 + 200000 + 100000 + 200000 + 100000 + 200000 + 100000 +
          200000 + 100000 + 200000 + 100000 + 200000 + 100000 + 200000 +
          100000 + 200000 + 100000 + 200000;
}
```

BlockIndent produces this: 

```console
$ clang-format -i -style="{BasedOnStyle: LLVM, AlignAfterOpenBracket: BlockIndent }" testfmt.c

int foo(void) {
  int x = (int
  )(100000 + 200000 + 100000 + 200000 + 100000 + 200000 + 100000 + 200000 +
    100000 + 200000 + 100000 + 200000 + 100000 + 200000 + 100000 + 200000 +
    100000 + 200000);
  int y = 100000 + 200000 + 100000 + 200000 + 100000 + 200000 + 100000 +
          200000 + 100000 + 200000 + 100000 + 200000 + 100000 + 200000 +
          100000 + 200000 + 100000 + 200000;
}
```

**Comment by gedare** [comment]
I proposed a fix at https://reviews.llvm.org/D137762, with this example it results in:
```
$clang-format -style="{BasedOnStyle: LLVM, AlignAfterOpenBracket: BlockIndent}" test.c

int foo(void) {
  int x = (int)(100000 + 200000 + 100000 + 200000 + 100000 + 200000 + 100000 +
                200000 + 100000 + 200000 + 100000 + 200000 + 100000 + 200000 +
                100000 + 200000 + 100000 + 200000);
  int y = 100000 + 200000 + 100000 + 200000 + 100000 + 200000 + 100000 +
          200000 + 100000 + 200000 + 100000 + 200000 + 100000 + 200000 +
          100000 + 200000 + 100000 + 200000;
}
```

**Comment by owenca** [comment]
Fixed in b40e9dce0a67.
